rsqueakvm/plugins/immutability_plugin.py

```diff
@@ -75,18 +75,20 @@ def primitiveImmutableFromArgs(interp, s_frame, argcount):
     :returns: An immutable object.
     :raises: PrimitiveFailedError
     """
+    if argcount == 0:
+        raise PrimitiveFailedError
     w_arguments = s_frame.pop_and_return_n(argcount)[:]
+    w_first_arg = w_arguments[0]
     w_cls = s_frame.pop()
     space = interp.space
     instance_kind = w_cls.as_class_get_shadow(space).get_instance_kind()
 
     if instance_kind == POINTERS:
         cls = select_immutable_pointers_class(w_arguments)
         return cls(space, w_cls, w_arguments)
-    # TBD:
-    # elif instance_kind == BYTES and isinstance(w_obj, W_BytesObject):
-    #     return W_Immutable_BytesObject(space, w_cls, w_obj.bytes)
-    # elif instance_kind == WORDS and isinstance(w_obj, W_WordsObject):
-    #     return W_Immutable_WordsObject(space, w_cls, w_obj.words)
+    elif instance_kind == BYTES and isinstance(w_first_arg, W_BytesObject):
```

Comment:
It would be rather be like
`ImmutableByteArray with: 65 with: 66 with: 68: with: 0 with 255`